src/item/enchantment/VanillaEnchantments.php

```diff
@@ -59,47 +60,181 @@ final class VanillaEnchantments{
 	use RegistryTrait;
 
 	protected static function setup() : void{
-		self::register("PROTECTION", new ProtectionEnchantment(KnownTranslationFactory::enchantment_protect_all(), Rarity::COMMON, ItemFlags::ARMOR, ItemFlags::NONE, 4, 0.75, null));
-		self::register("FIRE_PROTECTION", new ProtectionEnchantment(KnownTranslationFactory::enchantment_protect_fire(), Rarity::UNCOMMON, ItemFlags::ARMOR, ItemFlags::NONE, 4, 1.25, [
-			EntityDamageEvent::CAUSE_FIRE,
-			EntityDamageEvent::CAUSE_FIRE_TICK,
-			EntityDamageEvent::CAUSE_LAVA
-			//TODO: check fireballs
-		]));
-		self::register("FEATHER_FALLING", new ProtectionEnchantment(KnownTranslationFactory::enchantment_protect_fall(), Rarity::UNCOMMON, ItemFlags::FEET, ItemFlags::NONE, 4, 2.5, [
-			EntityDamageEvent::CAUSE_FALL
-		]));
-		self::register("BLAST_PROTECTION", new ProtectionEnchantment(KnownTranslationFactory::enchantment_protect_explosion(), Rarity::RARE, ItemFlags::ARMOR, ItemFlags::NONE, 4, 1.5, [
-			EntityDamageEvent::CAUSE_BLOCK_EXPLOSION,
-			EntityDamageEvent::CAUSE_ENTITY_EXPLOSION
-		]));
-		self::register("PROJECTILE_PROTECTION", new ProtectionEnchantment(KnownTranslationFactory::enchantment_protect_projectile(), Rarity::UNCOMMON, ItemFlags::ARMOR, ItemFlags::NONE, 4, 1.5, [
-			EntityDamageEvent::CAUSE_PROJECTILE
-		]));
-		self::register("THORNS", new Enchantment(KnownTranslationFactory::enchantment_thorns(), Rarity::MYTHIC, ItemFlags::TORSO, ItemFlags::HEAD | ItemFlags::LEGS | ItemFlags::FEET, 3));
-		self::register("RESPIRATION", new Enchantment(KnownTranslationFactory::enchantment_oxygen(), Rarity::RARE, ItemFlags::HEAD, ItemFlags::NONE, 3));
+		self::register("PROTECTION", new ProtectionEnchantment(KnownTranslationFactory::enchantment_protect_all(), Rarity::COMMON,
+			Flags::ARMOR,
+			Flags::NONE,
+			4,
+			0.75, null,
+			false,
+			fn($level) => 11 * ($level - 1) + 1,
```

Comment:
Inverting the tresor and min/max costs would seem to be the best way to remove the false in most cases.